Autopilot/Inc/TelemDriver.hpp

```diff
@@ -0,0 +1,48 @@
+#ifndef TELEM_DRIVER_HPP
+#define TELEM_DRIVER_HPP
+
+
+#include <ctime>
+#include <cstdint>
+
+struct TelemToPMData {
+    struct TelemToPMData { //Coming to telem from ground base.
+        float gpsLattitude, gpsLongitude; //Commanded gps position
+        float heading, distance //Commanded heading/distance
+        float roll, pitch, yaw;	//Commanded orientation (radians)
+        bool beginLand; //Command to start landing process
+        bool beginTakeoff; //Starting command
+        bool disconnectAutopilot; //Command to switch off autopilot
+    };
+};
+
+class TelemManager { //Pulling data from ground
+public:
+    virtual void GetResult(TelemToPMData* Data) = 0;
+};
+
+class XBEE : public TelemManager {
```

Comment:
For the Xbee, your'e going to want to treat it like any other sensor. That means it'll need its own header and implementation files. And one of the states in your state machine will need to ask it for data and give it data to send, which it will be doing periodically, as orchastrated by FREERTos. We'll talk mor about this as well